cmd/kube-controller-manager/app/options/options_test.go

```diff
@@ -113,169 +113,167 @@ func TestAddFlags(t *testing.T) {
 	f.Parse(args)
 	// Sort GCIgnoredResources because it's built from a map, which means the
 	// insertion order is random.
-	sort.Sort(sortedGCIgnoredResources(s.Generic.GarbageCollectorController.GCIgnoredResources))
+	sort.Sort(sortedGCIgnoredResources(s.GarbageCollectorController.GCIgnoredResources))
 
 	expected := &KubeControllerManagerOptions{
```

Comment:
how can we make sure that the ones of cloud and kube ctrl mgr do not diverge, where they shouldn't? Integration test?

Reply:
Others comments updated, but little confused about this, you mean `s.GarbageCollectorController.GCIgnoredResources`, for `GCIgnoredResources`'s memory address may be the same?

Reply:
I want to remove unrelated var which not belong to `CloudControllerManagerOptions ` struct in `cloud-controller-manager`'s `option_test.go`. But in others PRs about code clean, WDYT?

Reply:
I think my comment about `NewGenericComponentConfigOptions` solves this. Now they use the same functions, and the same defaulter.

Reply:
got it